diff --git a/scripts/mod/modpost.c b/scripts/mod/modpost.c
index c4e7d15..c0a814c 100644
--- a/scripts/mod/modpost.c
+++ b/scripts/mod/modpost.c
@@ -1572,7 +1572,9 @@ static void section_rela(const char *modname, struct elf_info *elf,
 	for (rela = start; rela < stop; rela++) {
 		r.r_offset = TO_NATIVE(rela->r_offset);
 #if KERNEL_ELFCLASS == ELFCLASS64
-		if (elf->hdr->e_machine == EM_MIPS) {
+#define EM_RISCV 243 
+		if (elf->hdr->e_machine == EM_MIPS ||
+		    elf->hdr->e_machine == EM_RISCV) {
 			unsigned int r_typ;
 			r_sym = ELF64_MIPS_R_SYM(rela->r_info);
 			r_sym = TO_NATIVE(r_sym);
